org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/ScalaJavaDepWhenJavaIsWrongTest.scala

```diff
@@ -0,0 +1,83 @@
+package org.scalaide.core
+package sbtbuilder
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.junit.Assert._
+import org.junit.Before
+import org.junit.Test
+
+object ScalaJavaDepWhenJavaIsWrongTest extends testsetup.TestProjectSetup("scalajavadepjavaiswrong")
+
+class ScalaJavaDepWhenJavaIsWrongTest {
+  import ScalaJavaDepWhenJavaIsWrongTest._
+  import testsetup.SDTTestUtils._
+
+  @Before def setup(): Unit = {
+    enableAutoBuild(false)
+  }
+
+  @Test def shouldCreateProjectFromWrongJava(): Unit = {
+    val TypeCMustImplementFoo = 1
+    val SIsNotAType = 1
+    val BarIsNotMemberOfJ = 1
+    cleanProject()
+
+    val aClass = compilationUnit("ticket_1000607/A.scala")
+    val cClass = compilationUnit("ticket_1000607/C.java")
+    val JJavaCU = compilationUnit("test/J.java")
+    val SScalaCU = compilationUnit("test/S.scala")
+    def findProblems = getProblemMarkers(aClass, cClass, JJavaCU, SScalaCU)
+
+    rebuild()
+    val problems0 = findProblems
+    assertTrue("One build error expected, got: " + userFriendlyMarkers(problems0), problems0.length == TypeCMustImplementFoo)
```

Comment:
That sounds good in theory, but it might prove hard to do if the error message is different in two different versions of the Scala compiler (2.12-M1 is around the corner).